### PR TITLE
Order Detail: Customer information section redesign

### PR DESCRIPTION
**Pull request by pmusolino**
Implemented the new design for Customer Information in Order Details, following the Refunds i3 designs.

This PR is related to this issue: Closes https://github.com/woocommerce/woocommerce-ios/issues/1091

After this issue, will be implemented the screen for Billing information https://github.com/woocommerce/woocommerce-ios/issues/1093

## Changes
- Added new cell `CustomerNoteTableViewCell` for showing the customer's note.
- Removed `ShowHideSectionFooter` which now will be unused.
- Edited `OrderDetailsDataSource` to manage the new sections.
- Edited `OrderDetailsViewModel` removing unused code for billing section.
- Edited `OrderDetailsViewController` to remove the old code about billing section.
- Used the `WooBasicTableViewCell` for creating the footer billing button in Customer section. The action needs to be connected to the new view, which will be implemented in a new PR.
- Implemented `CustomerNoteTableViewCell` tests
- Implemented `CustomerInfoTableViewCell` tests

## Testing

- Look at the code
- Run the app, and open an order detail which contains customer details and customer's note.
- Run WooCommerceTests

## Screenshots

| Customer detail without note             |  Customer detail with note |
:-------------------------:|:-------------------------:
Simulator Screen Shot - iPhone Xs - 2019-08-26 at 14 03 55 | Simulator Screen Shot - iPhone Xs - 2019-08-26 at 14 03 45

### CODE REVIEW

**Comment by pmusolino** [comment]
Ready for review @ctarda & @kyleaparker👍
Thanks!

File: WooCommerce/Classes/ViewModels/OrderDetailsViewModel/OrderDetailsDataSource.swift [hunk 1]
```diff
@@ -217,68 +180,30 @@ private extension OrderDetailsDataSource {
...
 
-        cell.isAccessibilityElement = true
         cell.accessibilityTraits = .button
-        cell.accessibilityLabel = String.localizedStringWithFormat(
-            NSLocalizedString(
-                "Phone number: %@",
-                comment: "Accessibility label that lets the user know the data is a phone number before speaking the phone number."
-            ),
-            phoneNumber
+        cell.accessibilityLabel = NSLocalizedString(
```
Comment (ctarda) [inline WooCommerce/Classes/ViewModels/OrderDetailsViewModel/OrderDetailsDataSource.swift]: I can not find the exact link to the accessibility guidelines, but as far as I recall the label should not include the type of control. There is a mention to it in the accessibility section of UIControl: https://developer.apple.com/documentation/uikit/uicontrol

Comment (pmusolino) [reply WooCommerce/Classes/ViewModels/OrderDetailsViewModel/OrderDetailsDataSource.swift]: I used the Accessibility Inspector to check, and when I Run an Audit, I get some warning about this issue, not only in this specific case but all the accessibility labels used inside this class.
I'm going to fix them. 👍

**Comment by pmusolino** [comment]
Ready for a new review @ctarda 👍

**Pull request merged**